lib/algo/DMD.h

```diff
@@ -258,6 +283,36 @@ class DMD
      */
     std::vector<Vector*> d_sampled_times;
 
+    /**
+     * @brief Whether to use init state as offset.
+     */
+    bool d_init_os_s = false;
+
+    /**
+     * @brief Whether to use init derivative as offset in NonuniformDMD.
```

Comment:
Should `NonuniformDMD` be mentioned in the base class?

Reply:
Both AdaptiveDMD and Nonuniform DMD are derived classes of DMD, and there are some options/variables that only apply to AdaptiveDMD (and vanilla DMD) or Nonuniform DMD. I think adding some comments to distinguish them will make good references to further development/usage.

Reply:
If there is a way to avoid mentioning derived classes in the base class, I think that would be preferable. A variable should be in the base class only if it is potentially used in general for multiple derived classes. If it is only intended for one derived class, as this comment seems to suggest, then it should be a member of the derived class, not the base class. Is there a better way to design this, by putting the variable in the derived class? Or is it a general variable, so that there should be no mention of `NonuniformDMD`, i.e. just comment `Whether to use init derivative as offset.`

Reply:
Now `d_derivative_offset` is only in `NonuniformDMD` class, and a new function `addOffset` is added.